apollo-api/src/main/java/com/apollographql/apollo/api/Field.java

```diff
@@ -103,6 +103,21 @@ public String cacheKey(Operation.Variables variables) {
     return String.format("%s(%s)", fieldName(), orderIndependentKey(arguments, variables));
   }
 
+  @SuppressWarnings("unchecked") public Object resolveArgument(String name, Operation.Variables variables) {
+    Map<String, Object> variableValues = variables.valueMap();
+    Object argumentValue = arguments.get(name);
+    if (argumentValue instanceof Map) {
+      Map<String, Object> argumentValueMap = (Map<String, Object>) argumentValue;
+      if (isArgumentValueVariableType(argumentValueMap)) {
+        String variableName = argumentValueMap.get(VARIABLE_NAME_KEY).toString();
+        return variableValues.get(variableName);
+      } else {
+        return null;
```

Comment:
I would love to have Optional here, but one concern is that this is public API so if user uses Guava for in his application won't it be confusing for him to deal with 2 different implementations of Optional, guava and our version?